### ISSUE TITLE
It doesn't work on Android 9.x and higher

### ISSUE THREAD

**Issue by yong-asial** [alan-ai/alan-sdk-cordova#4]
It seems to work on Android 8.x but not Android 9.x or higher.

Do you have any plan to support this?

Or any trick to get around it?

**Comment by snyuryev** [comment]
@yong-asial AlanSDK works with Android 9.x and higher. Could you provide additional information about your issue?

**Comment by yong-asial** [comment]
@snyuryev Thank you for the response.
Here is the build logs.

# Android 10.x
- It complaints about `android-minSdkVersion`
```
[15:56:10] > Task :app:processDebugMainManifest FAILED
[15:56:10] See http://g.co/androidstudio/manifest-merger for more information about the manifest merger.
[15:56:10] > Task :app:mergeDebugResources
[15:56:10] /tmp/download/platforms/android/app/src/main/AndroidManifest.xml Error:
[15:56:10] 	uses-sdk:minSdkVersion 21 cannot be smaller than version 22 declared in library [:CordovaLib] /tmp/download/platforms/android/CordovaLib/build/intermediates/merged_manifest/debug/AndroidManifest.xml as the library might be using APIs not available in 21
[15:56:10] 	Suggestion: use a compatible library with a minSdk of at most 21,
[15:56:10] 		or increase this project's minSdk version to at least 22,
[15:56:10] 		or use tools:overrideLibrary="org.apache.cordova" to force usage (may lead to runtime failures)
[15:56:10] Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
[15:56:10] You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
[15:56:10] See https://docs.gradle.org/7.1.1/userguide/command_line_interface.html#sec:command_line_warnings
[15:56:10] 14 actionable tasks: 14 executed
[15:56:10] FAILURE: Build failed with an exception.
[15:56:10] * What went wrong:
[15:56:10] Execution failed for task ':app:processDebugMainManifest'.
[15:56:10] > Manifest merger failed : uses-sdk:minSdkVersion 21 cannot be smaller than version 22 declared in library [:CordovaLib] /tmp/download/platforms/android/CordovaLib/build/intermediates/merged_manifest/debug/AndroidManifest.xml as the library might be using APIs not available in 21
[15:56:10]   	Suggestion: use a compatible library with a minSdk of at most 21,
[15:56:10]   		or increase this project's minSdk version to at least 22,
[15:56:10]   		or use tools:overrideLibrary="org.apache.cordova" to force usage (may lead to runtime failures)
[15:56:10] * Try:
[15:56:10] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[15:56:10] * Get more help at https://help.gradle.org
[15:56:10] BUILD FAILED in 21s
[15:56:10] Command failed with exit code 1: /tmp/download/platforms/android/gradlew cdvBuildDebug -b /tmp/download/platforms/android/build.gradle
```
[build-log-1655967376.log](https://github.com/alan-ai/alan-sdk-cordova/files/8966123/build-log-1655967376.log)


# Android 8.x
- working fine
[build-log-1655966960.log](https://github.com/alan-ai/alan-sdk-cordova/files/8966136/build-log-1655966960.log)

Also, if you notice, in the Github example that you provided https://github.com/alan-ai/alan-sdk-cordova/blob/master/examples/AlanBasicExample/package.json#L17 is Android8.x. So i just thought maybe you haven't tested with higher android platform yet. (while cordova-ios is 6.2)

**Comment by snyuryev** [comment]
@yong-asial 
```"cordova-android": "^8.1.0"``` is cordova plugin version. It doesn't reflect Android OS version.
What do you have as ```minSdkVersion``` for you app in ```build.gradle```?

**Comment by yong-asial** [comment]
@snyuryev Thank you. So, let me rephrase the question. Does it support for "cordova-android": "^10.x"?

**Comment by snyuryev** [comment]
@yong-asial Yes. Still have the same question:
What do you have as ```minSdkVersion``` for you app in ```build.gradle```?

**Comment by yong-asial** [comment]
@snyuryev No, i don't explicitly set any `minSdkVersion`. It should be the default value from either Cordova version or Cordova-Android version.

> What do you have as minSdkVersion for you app in build.gradle?

Sorry, I don't know because I use a cloud service to build so i could only get the build log (as i sent above).

Here is the result:
- Cordova 9.x and Cordova-Android 8.x: Work 
- Cordova 11.x and Cordova-Android 10.x: Not Work 
   - regardless of setting `<preference name="android-minSdkVersion" value="22" />` in the `config.xml`

**Comment by snyuryev** [comment]
@yong-asial Can you share ```build.gradle``` from ```app``` folder please?

**Comment by yong-asial** [comment]
Sorry, I don't have. The build log is all i have.

**Comment by snyuryev** [comment]
It should be located inside platforms->android->app

**Comment by yong-asial** [comment]
Sorry, I think i mistakenly close the ticket. I have resolved the issue by setting `minSdkVersion` to `21`.

What i found out is
The default `minSdkVersion` for Cordova 10, Cordova 11 as well as Cordova-android 9 and Cordova-android 10 is `22` -> that's why it wasn't worked with the plugin since the plugin require `minSdkVersion` to be `21`.

The reason why it worked for Cordova 9 and Cordova-Android 8 is because the default value of `minSdkVersion` is `21` -> it's compatible with the plugin.

So my fix to make it work for Cordova-Android 9.x and higher is to set `minSdkVersion` to `21` so that it's compatible with plugin.

@snyuryev Do you have any plan to upgrade the `minSdkVersion` of the plugin to higher version?


Sorry, again, it's not that I don't want to send you the build.gradle but the problem is the cloud service I'm using only return me the build log. All cordova-related commands (cordova platform add, cordova plugin add, cordova prepare, ...) are all done in the server. So in my local project, I don't have any folders related to Cordova such as plugins, platform, etc.

![Screen Shot 2022-06-24 at 17 20 51](https://user-images.githubusercontent.com/28589278/175494766-cf279f22-5113-4418-93b6-6c929039d574.png)

**Comment by snyuryev** [comment]
@yong-asial we don't have plans to update ```minSdkVersion``` - that will break old versions of cordova plugins.
Manual setup ```minSdkVersion``` is absolutely usual situation.

**Comment by yong-asial** [comment]
@snyuryev Thank you for the reply.
